packages/nextjs/components/scaffold-eth/Contract/utilsContract.tsx

```diff
@@ -18,14 +18,21 @@ const getFunctionInputKey = (functionName: string, input: AbiParameter, inputInd
  * @param e - error object
  * @returns {string} parsed error string
  */
-const getParsedError = (e: any | BaseViemError): string => {
-  let message = e.message ?? "An unknown error occurred";
-
+const getParsedError = (e: any): string => {
+  let message: string = e.message ?? "An unknown error occurred";
   if (e instanceof BaseViemError) {
     if (e.details) {
       message = e.details;
     } else if (e.shortMessage) {
       message = e.shortMessage;
+      const cause = e.cause as any as { data?: DecodeErrorResultReturnType };
+      // if its not generic error, append custom error name and its args to message
+      if (cause.data && cause.data?.abiItem?.name !== "Error") {
+        const customErrorArgs = cause.data.args?.toString() ?? "";
+        message = `${message.replace(/reverted\.$/, "reverted with following reason:")}\n${
```

Comment:
example message: `The contract function "withdraw" reverted.`

Since the message contains a double quote, had to use `/reverted\.$/` regex, simply doing `message.replace("reverted.", "...")` doesn't work. 

Am I missing something? 

```ts

// 
message.replace("reverted." , "...")

```

Reply:
Ohh yeah it works with `.replace("reverted.")` , actually my initial logic while debugging was a bit messier and might have made a mistake there because of which `.replace("reverted.")` was not working I thought it was due to presence of double quotes in `message` but I am wrong here

> I like your approach, where you check that the "reverted." string is at the end of the message.

TBH it was github copilot :), but yeah checking "reverted." is very end of string make more sense so keeping regex 🙌

Reply:
> TBH it was github copilot :)

Like 90% of my coding these days haha